packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -1434,6 +1478,24 @@ export class LocalDetachedFluidDataStoreContext
 		return this.channelToDataStoreFn(await this.channelP);
 	}
 
+	/**
+	 * This method provides a synchronous path for binding a runtime to the context.
+	 *
+	 * Due to its synchronous nature, it is unable to validate that the runtime
+	 * represents a datastore which is instantiable by remote clients. This could
+	 * happen if the runtime's package path does not return a factory when looked up
+	 * in the container runtime's registry, or if the runtime's entrypoint is not
+	 * properly initialized. As both of these validation's are asynchronous to preform.
+	 *
+	 * If used incorrectly, this function can result in permanent data corruption.
+	 */
+	public unsafe_AttachRuntimeSync(channel: IFluidDataStoreChannel) {
```

Comment:
Shouldn't it be protected?

Reply:
Only context itself calls into it, so it should be possible to hide it.

Reply:
this class isn't exported, it only used within this package. it can't be anything, but public given the current class hierarchy, but again, it's not visible outside this package. the comment is for fluid framework developers.